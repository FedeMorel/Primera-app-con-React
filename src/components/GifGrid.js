import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GIfGridItem';



const GifGrid = ({category}) =>{
    
    const {data, loading} = useFetchGifs(category);
    
    return(
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading</p>}
            <div className="content">
                {data.map( img => <GifGridItem key={img.id} {...img} />)}
            </div>
        </>
    );
}

GifGrid.protoTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;