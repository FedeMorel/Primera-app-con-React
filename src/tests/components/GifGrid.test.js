import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom'
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');
    const category = 'batman'


describe('Pruebas en <GifGrid/>', ()=>{


    test('Debe demostarse correctamente el componente', ()=>{
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe demostrar items cuando se cargan imÃ¡genes con useFetchGifs', ()=>{

        const gifs=[{
            id: 'ABC',
            img: 'https://localhost/url/prueba.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        expect(wrapper).toMatchSnapshot();
    });

});