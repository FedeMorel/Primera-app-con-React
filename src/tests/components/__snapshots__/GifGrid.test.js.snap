// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe demostarse correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    batman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="content"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid/> Debe demostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    batman
  </h3>
  <div
    className="content"
  >
    <GifGridItem
      id="ABC"
      img="https://localhost/url/prueba.jpg"
      key="ABC"
      title="Cualquier cosa"
    />
  </div>
</Fragment>
`;
